### PR TITLE
feat(staking): add governance ui methods

### PR DESCRIPTION
**Pull request by keyvankhademi**

### CODE REVIEW

File: governance/pyth_staking_sdk/src/pyth-staking-client.ts [hunk 1]
```diff
@@ -710,4 +716,44 @@ export class PythStakingClient {
...
+      );
+    return convertBNToBigInt(targetAccount);
+  }
+
+  /**
+   * This returns the current scaling factor between staked tokens and realms voter weight.
+   * The formula is n_staked_tokens = scaling_factor * n_voter_weight
+   */
+  public async getScalingFactor(): Promise<number> {
+    const targetAccount = await this.getTargetAccount();
```
Comment (guibescos) [inline governance/pyth_staking_sdk/src/pyth-staking-client.ts:733]: Note: this is only valid if someone has cranked the target account (happens whenever someone stakes or unstakes)

File: governance/pyth_staking_sdk/src/pyth-staking-client.ts [hunk 2]
```diff
@@ -710,4 +716,44 @@ export class PythStakingClient {
...
+  }
+
+  public async getUpdatePoolAuthorityInstruction(
+    governanceAuthority: PublicKey,
+    poolAuthority: PublicKey,
+  ): Promise<TransactionInstruction> {
+    return this.stakingProgram.methods
+      .updatePoolAuthority(poolAuthority)
+      .accounts({
+        governanceAuthority,
```
Comment (guibescos) [inline governance/pyth_staking_sdk/src/pyth-staking-client.ts:755]: Why doesn't governanceAuthority get inferred here and it does get inferred in recoverAccount

**Review by guibescos — APPROVED** [review]

File: governance/pyth_staking_sdk/src/pyth-staking-client.ts (hunk 2 above)
Comment (keyvankhademi) [reply governance/pyth_staking_sdk/src/pyth-staking-client.ts:735]: fixed.

File: governance/pyth_staking_sdk/src/pyth-staking-client.ts [hunk 3]
```diff
@@ -687,4 +694,112 @@ export class PythStakingClient {
...
+
+  public async getVoterWeight() {
+    const mainAccount = await this.getMainStakeAccount();
+
+    if (mainAccount === undefined) {
+      return 0;
+    }
+
+    const scalingFactor = await this.getScalingFactor();
+    return Number(mainAccount.balance) / scalingFactor;
```
Comment (guibescos) [inline governance/pyth_staking_sdk/src/pyth-staking-client.ts:803]: It needs to be the voter weight, that is the sum of all positions that have governance target and are active

File: governance/pyth_staking_sdk/src/pyth-staking-client.ts [hunk 4]
```diff
@@ -687,4 +694,112 @@ export class PythStakingClient {
...
+
+    if (mainAccount === undefined) {
+      return;
+    }
+
+    for (let i = 1; i < stakeAccountBalances.length; i++) {
+      const currentAccount = stakeAccountBalances[i];
+      if (
+        currentAccount !== undefined &&
+        currentAccount.balance > mainAccount.balance
```
Comment (guibescos) [inline governance/pyth_staking_sdk/src/pyth-staking-client.ts:786]: Sort by voterWeight not by balance

**Review by guibescos — COMMENTED** [review]
I think you also need a method to return the user's voter weight account and the total voter weight account

File: governance/pyth_staking_sdk/src/pyth-staking-client.ts (hunk 3 above)
Comment (keyvankhademi) [reply governance/pyth_staking_sdk/src/pyth-staking-client.ts:803]: right!

**Pull request merged**